azurerm/data_source_loadbalancer.go

```diff
@@ -0,0 +1,210 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-04-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmLoadBalancer() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmLoadBalancerRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"location": locationForDataSourceSchema(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"frontend_ip_configuration": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"subnet_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"private_ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"public_ip_address_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"private_ip_address_allocation": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"load_balancer_rules": {
+							Type:     schema.TypeSet,
```

Comment:
How are you anticipating people will reference this? The interpolation would need the value before it could know the key to reference it by, and I think count-based loops don't actually work.

Reply:
no because the ordering will change outside of this, it's unlikely this field will be referenced so we should be able to just remove it for the moment